src/example-types/results.js

```diff
@@ -17,34 +27,26 @@ let convertPopulate = getSchema =>
         )
 
       let $lookup = {
-        $lookup: include
-          ? {
-              as,
-              from: targetCollection,
-              let: { localField: `$${localField}` },
-              pipeline: [
-                {
-                  $match: {
-                    $expr: { $eq: [`$${foreignField}`, '$$localField'] },
-                  },
-                },
-                { $project: projectFromInclude(include) },
-              ],
-            }
-          : {
-              as,
-              from: targetCollection,
-              localField,
-              foreignField, // || node.schema, <-- needs schema lookup
-            },
+        $lookup: {
+          as,
+          from: targetCollection,
+          localField,
+          foreignField, // || node.schema, <-- needs schema lookup
+        },
       }
+
+      let $project = include
+        ? { $project: omitFromInclude(targetSchema, include, as) }
```

Comment:
Why can't we use `projectFromInclude` here and have to omit instead?

Reply:
@stellarhoof Because we are only referring to a single populated object on the entire node for each of these lookups.  So if we did an include instead of an omit (without hard coding the default node props we want to keep) we would be ignoring all of the sibling properties on the node that we might actually want to keep that have nothing to do with the populated stuff.  I asked @doug the same question.

Reply:
I don't follow. This is used to build mongo queries to get results and counts. It has nothing to do with the contexture node's properties from what I can see.

Reply:
@stellarhoof - the slow form of the query is the one that uses `let` and `pipeline` -- it takes several seconds while the `foreignField` form takes a couple milliseconds.

To avoid the let-pipeline form, I have to to the $lookup first, then `project: { field: 0 }` the resulting fields. Think it through -- there's no other way to do that.

Reply:
I understand we need to avoid the lookup pipeline. I'm just not understanding what difference is there between a whitelist/blacklist here. It seems to me it can be done either way.

Reply:
Sorry, just realized there's one thing to make that clearer @stellarhoof, my bad -- the issue is that you don't know what properties the base record has other than the `as` property, so any kind of `project: { field: 1 }` is going to need the base schema with its fields in scope so you can make sure to keep them with `project: { field: 1 }` -- but it's not in scope to my knowledge

Reply:
or does `getSchema` give me that? I'll look...

Reply:
It seems mongo will ignore any bogus properties. Ex:

```
db.getCollection('user').aggregate([
{ $match: {email: 'ahernandez@govspend.com'} },
{ $project: { email: 1, foo: 1 } }
])
```

Will not include `foo`

> any kind of project: { field: 1 } is going to need the base schema with its fields in scope so you can make sure to keep them with project: { field: 1 }

The fields to keep are already determined by the `include` list. Am I missing something?

Reply:
@stellarhoof that's not to the worry, the worry is that we know that our records have `as`, but we don't know any of their other props

Reply:
what I can't figure out quickly is whether there's a decent way to get the base schema in scope -- if I could do that I could do it with the includes directly, granted

Reply:
@stellarhoof OK, looks like I can pass in a second `localSchema` argument at https://github.com/smartprocure/contexture-mongo/blob/master/src/example-types/results.js#L4 and we'll have the value for that from https://github.com/smartprocure/contexture-mongo/blob/master/src/example-types/results.js#L125

However, I don't think we should do this because it will make the queries larger, sometime significantly larger. Omitting makes for a smaller project stage in the query and will perform better.

Not making any changes.

@daedalus28 ready for review

Reply:
Even though I don't like this change I'm going to go ahead and make it to move this thing through.

Reply:
Actually, having thought about it I don't even not like it -- the query difference won't be much and in some cases the queries will get smaller, too.

Reply:
OK, after further discussion we'll leave as-is and i'll put in a comment

Reply:
resolved. thanks for discussion @stellarhoof and @ccalbaugh !